diff --git a/src/modules/acc/acc_cdr.c b/src/modules/acc/acc_cdr.c
index 6a7862dd74..42462b6304 100644
--- a/src/modules/acc/acc_cdr.c
+++ b/src/modules/acc/acc_cdr.c
@@ -389,6 +389,16 @@ static int write_cdr( struct dlg_cell* dialog,
 		return -1;
 	}
 
+	/* Skip cdr if cdr_skip dlg_var exists */
+	if (cdr_skip.len > 0) {
+		str* nocdr_val = 0;
+		nocdr_val = dlgb.get_dlg_var( dialog, &cdr_skip);
+		if ( nocdr_val ){
+			LM_DBG( "cdr_skip dlg_var set, skip cdr!");
+			return 0;
+		}
+	}
+
 	ret = log_write_cdr(dialog, message);
 	ret |= db_write_cdr(dialog, message);
 	return ret;
diff --git a/src/modules/acc/acc_mod.c b/src/modules/acc/acc_mod.c
index 959bd8b0ca..77971b04f2 100644
--- a/src/modules/acc/acc_mod.c
+++ b/src/modules/acc/acc_mod.c
@@ -121,6 +121,7 @@ int cdr_start_on_confirmed = 0;
 int cdr_expired_dlg_enable = 0;
 static char* cdr_facility_str = 0;
 static char* cdr_log_extra_str = 0;
+str cdr_skip = {NULL, 0};
 
 str cdr_start_str = str_init("start_time");
 str cdr_end_str = str_init("end_time");
@@ -201,6 +202,7 @@ static param_export_t params[] = {
 	{"log_extra",            PARAM_STRING, &log_extra_str        },
 	/* cdr specific */
 	{"cdr_enable",           INT_PARAM, &cdr_enable                 },
+	{"cdr_skip",             PARAM_STR, &cdr_skip                   },
 	{"cdr_log_enable",         INT_PARAM, &cdr_log_enable           },
 	{"cdr_start_on_confirmed", INT_PARAM, &cdr_start_on_confirmed   },
 	{"cdr_facility",         PARAM_STRING, &cdr_facility_str           },
diff --git a/src/modules/acc/acc_mod.h b/src/modules/acc/acc_mod.h
index a28721cdea..b25c9d4a17 100644
--- a/src/modules/acc/acc_mod.h
+++ b/src/modules/acc/acc_mod.h
@@ -53,6 +53,7 @@ extern int cdr_extra_nullable;
 extern int cdr_start_on_confirmed;
 extern int cdr_log_facility;
 extern int cdr_expired_dlg_enable;
+extern str cdr_skip;
 
 extern int db_flag;
 extern int db_missed_flag;
